### PR TITLE
test: Refactors GetClusterInfo

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Changes:
- rename ClusterInfo fields <https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2414#discussion_r1676021716>
- support a data source when using a cluster via env-vars
  - manual test: `PASS: TestAccBackupRSCloudBackupSnapshot_basic (212.21s)` (used `990.59s` in CI)
- cleanup `acc.advanced_cluster.go` (moved `ConfigClusterGlobal` to resource_cluster_test.go)
- move out ClusterResourceHcl to `config_cluster.go`
- add documentation to ClusterInfo fields and methods : <https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2412#discussion_r1677514004>
- remove `ClusterInfo.ProjectIDStr` this field was not safe to use when a reference (e.g., `mongodbatlas_project.test.id`)  was used for the project id.

Link to any related issue(s): CLOUDP-260997

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/cloudbackupschedule/resource_cloud_backup_schedule_test.go [hunk 1]
```diff
@@ -676,16 +676,16 @@ func configAzure(info *acc.ClusterInfo, policy *admin.DiskBackupApiPolicyItem) s
...
+	`, info.TerraformNameRef, info.ProjectID, policy.GetFrequencyInterval(), policy.GetRetentionUnit(), policy.GetRetentionValue())
 }
 
 func configAdvancedPolicies(info *acc.ClusterInfo, p *admin.DiskBackupSnapshotSchedule) string {
-	return info.ClusterTerraformStr + fmt.Sprintf(`
+	return info.TerraformStr + fmt.Sprintf(`
 		resource "mongodbatlas_cloud_backup_schedule" "schedule_test" {
 			cluster_name     = %[1]s
-			project_id       = %[2]s
+			project_id       = %[2]q
```
Comment (lantoli) [inline internal/service/cloudbackupschedule/resource_cloud_backup_schedule_test.go:688]: just to confirm this change is ok

File: internal/service/cloudbackupschedule/resource_cloud_backup_schedule_test.go [hunk 2]
```diff
@@ -775,20 +775,20 @@ func configExportPolicies(info *acc.ClusterInfo, policyName, roleName, bucketNam
     }
 
     resource "mongodbatlas_cloud_provider_access_setup" "setup_only" {
-        project_id      = %[2]s
+        project_id      = %[2]q
```
Comment (lantoli) [inline internal/service/cloudbackupschedule/resource_cloud_backup_schedule_test.go:778]: it's ok to do quotes here and the rest?

File: internal/service/cloudbackupschedule/resource_cloud_backup_schedule_test.go (hunk 1 above)
Comment (EspenAlbert) [reply internal/service/cloudbackupschedule/resource_cloud_backup_schedule_test.go:688]: Yes, this is part of the (last bullet in description):
> remove ClusterInfo.ProjectIDStr this field was not safe to use when a reference (e.g., mongodbatlas_project.test.id) was used for the project id.

File: internal/testutil/acc/cluster.go [hunk 3]
```diff
@@ -10,6 +10,8 @@ import (
 	"go.mongodb.org/atlas-sdk/v20240530002/admin"
 )
 
+// ClusterRequest contains configuration for a cluster where all fields are optional and AddDefaults are used for required fields.
```
Comment (lantoli) [inline internal/testutil/acc/cluster.go]:
```suggestion
// ClusterRequest contains configuration for a cluster where all fields are optional and AddDefaults is used for required fields.
```

File: internal/service/cloudbackupschedule/resource_cloud_backup_schedule_test.go (hunk 2 above)
Comment (EspenAlbert) [reply internal/service/cloudbackupschedule/resource_cloud_backup_schedule_test.go:778]: See comment above: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2426#discussion_r1679179283

**Review by lantoli — APPROVED** [review]

**Comment by EspenAlbert** [comment]
Cancelled test run in 250f108 to avoid running all acceptance tests again (only docs change)
Waiting for tests in [cfd2a30](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2426/commits/cfd2a3069733a5f06c2e5e6b1eda0e07b59a5074) to complete (only [`TestAccBackupRSCloudBackupSchedule_basic`](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9954654668/job/27500963754#step:5:94) failed in [e31b7d6](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2426/commits/e31b7d6729acdca76230a5bc9194d51a9e1511df)

All tests are passing: https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9955340316/job/27503054147 ✅

**Pull request merged**